android/src/main/kotlin/com/bitmovin/player/flutter/FlutterPlayerView.kt

```diff
@@ -35,10 +37,33 @@ class FlutterPlayerView(
             messenger,
         )
 
+    private val activity = context.requireActivity()
+    private val playerView: PlayerView = PlayerView(context, player = null)
+
+    private val activityLifecycle =
+        activity.let { it as? FlutterActivity ?: it as? FlutterFragmentActivity }
+            ?.lifecycle
+            ?: error(
+                "Trying to create an instance of ${this::class.simpleName}" +
+                    " while not attached to a FlutterActivity or FlutterFragmentActivity",
+            )
```

Comment:
The section `Alternatives to FlutterActivity` [here](https://api.flutter.dev/javadoc/io/flutter/embedding/android/FlutterActivity.html) indicate that a Flutter view can be part of different types of activities. While this is probably not the common use case we could just use
```
(activity as? LifecycleOwner)
```
instead to be more resilient.

Reply:
I read through the docs and I don't expect any problems because of this. But I also see no issue in just using the LifecycleOwner observer cast, so I'll do that just to be safe.

[bd37436](https://github.com/bitmovin/bitmovin-player-flutter/pull/85/commits/bd3743674b5a61f27ec482f416739af81f28754d)